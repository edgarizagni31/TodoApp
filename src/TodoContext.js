import { createContext, useEffect, useReducer, useState } from 'react';
import { todoReducer } from './helpers/todoReducer';

export const TodoContext = createContext();

const  TodoProvider = ( {children} ) => {
    const init = () => JSON.parse(localStorage.getItem('todos')) || [];
    const [ todos, dispatch ] = useReducer( todoReducer, [], init );
    const [ listTodos, setListTodos ] = useState(todos);

    useEffect( 
        () => {
            localStorage.setItem('todos', JSON.stringify(todos));
            setListTodos( todos );
        }
    , [ todos ])
   
    const filterComplete =   () => {
      setListTodos( todos.filter( (t) => t.done === true ) );
    }
  
    const filterActives =  () => {
      setListTodos( todos.filter( (t) => t.done === false ) )
    }

    const resetTodos = () => {
      setListTodos( todos );
    }

    return (
        <TodoContext.Provider 
            value = {
                {
                    listTodos,
                    dispatch,
                    resetTodos,
                    filterActives,
                    filterComplete
                }
            }
        >
            {children}
        </TodoContext.Provider>
    )
}

export default TodoProvider;
