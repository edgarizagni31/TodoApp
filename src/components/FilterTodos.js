import React from 'react'
import { TodoListItem } from './TodoListItem'

const FilterTodos = ( { type, todos, deleteTodo, toogleTodo } ) => {

    return (
        <>
            {
                type === 'actives'
                ?
                todos.map( todo => (
                    !todo.done &&
                    <TodoListItem
                        key = {todo.id}
                        todo = {todo}
                        deleteTodo = {deleteTodo}
                        toogleTodo = {toogleTodo}
                    />
                ))
                :
                todos.map( todo => (
                    todo.done &&
                    <TodoListItem
                        key = {todo.id}
                        todo = {todo}
                        deleteTodo = {deleteTodo}
                        toogleTodo = {toogleTodo}
                    />
                ))
            }
        </>
    )
}

export default FilterTodos