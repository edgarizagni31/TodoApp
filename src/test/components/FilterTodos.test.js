import { mount } from 'enzyme';
import FilterTodos from '../../components/FilterTodos';
import { TodoContext } from '../../TodoContext';

describe('test to FilterTodos', () => {
    const resetTodos = jest.fn();
    const filterComplete = jest.fn();
    const filterActives = jest.fn();

    const wrapper = mount(
        <TodoContext.Provider value = { { resetTodos, filterComplete, filterActives } }>
            <FilterTodos />
        </ TodoContext.Provider>
    );

    test('should be displayed correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('FilterTodoItem').length ).toBe(3);
    });
})
