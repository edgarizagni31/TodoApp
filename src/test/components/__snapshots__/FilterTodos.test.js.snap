// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to FilterTodos should be displayed correctly 1`] = `
<div
  aria-label="filtrar todos por "
  className="todo-list__options"
>
  <button
    className="todo-list__option"
    onClick={[MockFunction]}
  >
    Todas
  </button>
  <button
    className="todo-list__option"
    onClick={[MockFunction]}
  >
    Activas
  </button>
  <button
    className="todo-list__option"
    onClick={[MockFunction]}
  >
    Completadas
  </button>
</div>
`;
